src/View/View.php

```diff
@@ -554,6 +554,11 @@ public function render($view = null, $layout = null)
             return;
         }
 
+        if ($layout !== null) {
+            $defaultLayout = $this->layout;
+            $this->layout = $layout;
+        }
```

Comment:
Why do you need to restore the layout after rendering? If before rendering you set `$this->layout = $layout`. Then the view template/elements can manipulate the property as needed. Then when rendering the layout we can use `$this->layout` instead of the local variable.